src/extensions/agent_modules/cloud_communication/communication_channel/mqtt_broker/src/adu_communication_channel.c

```diff
@@ -911,6 +709,10 @@ bool PerformChannelStateManagement()
                 s_commMgrState.mqttSettings.hostname,
                 s_commMgrState.mqttSettings.tcpPort,
                 s_commMgrState.mqttSettings.keepAliveInSeconds);
+            // Unrecoverable error. Agent configuration may be invalid.
```

Comment:
What is meant by unrecoverable here? If truly cannot recover, then perhaps it should fail hard; otherwise, what's the point?